compiler-core/src/type_/tests.rs

```diff
@@ -125,6 +127,90 @@ macro_rules! assert_no_warnings {
     };
 }
 
+macro_rules! assert_import_warning {
```

Comment:
Rather than making a new macro let's copy the existing pattern in some of the other macros and make it possible to supply another module that can be imported.